backend/src/schema/resolvers/reports.spec.js

```diff
@@ -1,521 +1,593 @@
-import { GraphQLClient } from 'graphql-request'
-import Factory from '../../seed/factories'
-import { host, login, gql } from '../../jest/helpers'
-import { getDriver, neode } from '../../bootstrap/neo4j'
 import { createTestClient } from 'apollo-server-testing'
 import createServer from '../.././server'
+import Factory from '../../seed/factories'
+import { gql } from '../../jest/helpers'
+import { getDriver, neode as getNeode } from '../../bootstrap/neo4j'
 
 const factory = Factory()
-const instance = neode()
+const instance = getNeode()
 const driver = getDriver()
 
-describe('report mutation', () => {
-  let reportMutation
-  let headers
-  let client
-  let variables
-  let createPostVariables
-  let user
+describe('report resources', () => {
+  let authenticatedUser, currentUser, mutate, query, moderator, abusiveUser
   const categoryIds = ['cat9']
-
-  const action = () => {
-    reportMutation = gql`
-      mutation($resourceId: ID!, $reasonCategory: ReasonCategory!, $reasonDescription: String!) {
-        report(
-          resourceId: $resourceId
-          reasonCategory: $reasonCategory
-          reasonDescription: $reasonDescription
-        ) {
-          createdAt
-          reasonCategory
-          reasonDescription
-          type
-          submitter {
-            email
-          }
-          user {
-            name
-          }
-          post {
-            title
-          }
-          comment {
-            content
-          }
+  const reportMutation = gql`
+    mutation($resourceId: ID!, $reasonCategory: ReasonCategory!, $reasonDescription: String!) {
+      report(
+        resourceId: $resourceId
+        reasonCategory: $reasonCategory
+        reasonDescription: $reasonDescription
+      ) {
+        createdAt
+        reasonCategory
+        reasonDescription
+        type
+        submitter {
+          email
+        }
+        user {
+          name
+        }
+        post {
+          title
+        }
+        comment {
+          content
         }
       }
-    `
-    client = new GraphQLClient(host, {
-      headers,
-    })
-    return client.request(reportMutation, variables)
+    }
+  `
+  const variables = {
+    resourceId: 'whatever',
+    reasonCategory: 'other',
+    reasonDescription: 'Violates code of conduct !!!',
   }
 
-  beforeEach(async () => {
-    variables = {
-      resourceId: 'whatever',
-      reasonCategory: 'other',
-      reasonDescription: 'Violates code of conduct !!!',
-    }
-    headers = {}
-    user = await factory.create('User', {
-      id: 'u1',
-      role: 'user',
-      email: 'test@example.org',
-      password: '1234',
-    })
-    await factory.create('User', {
-      id: 'u2',
-      role: 'user',
-      name: 'abusive-user',
-      email: 'abusive-user@example.org',
-    })
-    await instance.create('Category', {
-      id: 'cat9',
-      name: 'Democracy & Politics',
-      icon: 'university',
+  beforeAll(async () => {
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-10-21T08:19:57Z" title="Monday, October 21st 2019, 10:19:57 am +02:00">Oct 21, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -1,521 +1,599 @@
-import { GraphQLClient } from 'graphql-request'
-import Factory from '../../seed/factories'
-import { host, login, gql } from '../../jest/helpers'
-import { getDriver, neode } from '../../bootstrap/neo4j'
 import { createTestClient } from 'apollo-server-testing'
 import createServer from '../.././server'
+import Factory from '../../seed/factories'
+import { gql } from '../../jest/helpers'
+import { getDriver, neode as getNeode } from '../../bootstrap/neo4j'
 
 const factory = Factory()
-const instance = neode()
+const instance = getNeode()
 const driver = getDriver()
 
-describe('report mutation', () => {
-  let reportMutation
-  let headers
-  let client
-  let variables
-  let createPostVariables
-  let user
+describe('report resources', () => {
+  let authenticatedUser, currentUser, mutate, query, moderator, abusiveUser
   const categoryIds = ['cat9']
-
-  const action = () => {
-    reportMutation = gql`
-      mutation($resourceId: ID!, $reasonCategory: ReasonCategory!, $reasonDescription: String!) {
-        report(
-          resourceId: $resourceId
-          reasonCategory: $reasonCategory
-          reasonDescription: $reasonDescription
-        ) {
-          createdAt
-          reasonCategory
-          reasonDescription
-          type
-          submitter {
-            email
-          }
-          user {
-            name
-          }
-          post {
-            title
-          }
-          comment {
-            content
-          }
+  const reportMutation = gql`
+    mutation($resourceId: ID!, $reasonCategory: ReasonCategory!, $reasonDescription: String!) {
+      report(
+        resourceId: $resourceId
+        reasonCategory: $reasonCategory
+        reasonDescription: $reasonDescription
+      ) {
+        createdAt
+        reasonCategory
+        reasonDescription
+        type
+        submitter {
+          email
+        }
+        user {
+          name
+        }
+        post {
+          title
+        }
+        comment {
+          content
         }
       }
-    `
-    client = new GraphQLClient(host, {
-      headers,
-    })
-    return client.request(reportMutation, variables)
+    }
+  `
+  const variables = {
+    resourceId: 'whatever',
+    reasonCategory: 'other',
+    reasonDescription: 'Violates code of conduct !!!',
   }
 
-  beforeEach(async () => {
-    variables = {
-      resourceId: 'whatever',
-      reasonCategory: 'other',
-      reasonDescription: 'Violates code of conduct !!!',
-    }
-    headers = {}
-    user = await factory.create('User', {
-      id: 'u1',
-      role: 'user',
-      email: 'test@example.org',
-      password: '1234',
-    })
-    await factory.create('User', {
-      id: 'u2',
-      role: 'user',
-      name: 'abusive-user',
-      email: 'abusive-user@example.org',
-    })
-    await instance.create('Category', {
-      id: 'cat9',
-      name: 'Democracy & Politics',
-      icon: 'university',
+  beforeAll(() => {
```


could you please add a `await factory.cleanDatabase()`? 
@Tirokk has brought up, and I agree with him, that it is confusing for new developers to get failing tests the first time the test is run because the database has seeded data.

If we clean the database once before any tests are run, we can guarantee no constraint validation errors thrown.